base/channels.jl

```diff
@@ -318,15 +319,19 @@ end
 
 function put_buffered(c::Channel, v)
     lock(c)
+    did_buffer = false
     try
+        @atomic :monotonic c.length += 1
```

Comment:
```suggestion
        @atomic :monotonic c.length = c.length + 1 # just atomic store, not increment
```

we don't actually want atomic increment (which is often quite slow), but simply a monotonic store

Reply:
Ah yes, this is inside the lock... so confusing!

So (to check I understand) I think this means:
* We don't need an atomic load — no other thread can be storing to this in parallel
* For the same reasons, we don't need atomic increment
* We do require atomic (monotonic) store, as other threads may do an atomic load outside the lock

Reply:
I'll go through and update all the places that this pattern occurs.